pkg/provisioners/kafka/controller/channel/reconcile.go

```diff
@@ -37,10 +37,13 @@ import (
 const (
 	finalizerName = controllerAgentName
 
-	ArgumentNumPartitions = "NumPartitions"
-	DefaultNumPartitions  = 1
+	DefaultNumPartitions = 1
 )
 
+type channelArgs struct {
+	NumPartitions int32 `json:"NumPartitions,omitempty"`
```

Comment:
(FYI) You don't need to supply a name parameter here the name is the same as the public field name.